### DISCUSSION TITLE
New sample - formula bar with intellisense

### DISCUSSION THREAD

**Discussion by MikeStall** [microsoft/Power-Fx#86] [General]
We have a new sample available: a Power Fx formula bar in a webpage with intellisense support: 

https://github.com/microsoft/power-fx-host-samples/tree/main/Samples/WebDemo

**Comment by Felx-B** [comment]
Hi 👋,
Thank you for this awesome project !
Do you plan to release the source code of `@microsoft/power-fx-formulabar` ?
I'd like to use it in an Angular app

**Reply by MikeStall** [reply]
We will release the source but don't have an ETA yet. For now, you can still get the built product as an npm package. Although it's a react control.

**Comment by diwakar-jha3110** [comment]
Hi,
Thank you for this awesome poject,
Can we implement powerfx services in node project instead of dotnet (sample web demo).
I have scenario to use it in Angular with node server.

**Reply by MikeStall** [reply]
The PowerFx compiler and interpreter are implemented in C#, so  that can't run in node. 
But the formula bar that _displays_ and edits an expression is in TypeScript, so that can run as a react control. But the formula bar still needs to communicate with a C#-based service to provide intellisense, and the expressions ultimately run in a C# interpreter.

**Reply by diwakar-jha3110** [reply]
Thank You for the prompt help

**Comment by Felx-B** [comment]
Hi 👋
For those who are interested, I made an Angular implementation of the formular bar
Check it out :
https://github.com/Felx-B/ng-power-fx-formulabar

**Reply by MikeStall** [reply]
That's great!  Very nice.

**Reply by diwakar-jha3110** [reply]
Thank You  Felx-B awesome.

**Comment by jwjosefy** [comment]
Thank you Felix! Great work!

Em ter., 4 de jan. de 2022 às 20:02, Mike Stall ***@***.***>
escreveu:
